synapse/storage/events.py

```diff
@@ -1548,6 +1556,86 @@ def _store_redaction(self, txn, event):
             (event.event_id, event.redacts),
         )
 
+    @defer.inlineCallbacks
+    def _censor_redactions(self):
+        """Censors all redactions older than a month that haven't been censored.
+
+        By censor we mean update the event_json table with the redacted event.
+
+        Returns:
+            Deferred
+        """
+
+        if not self.hs.config.redaction_retention_period:
+            return
+
+        max_pos = yield self.find_first_stream_ordering_after_ts(
+            self._clock.time_msec() - self.hs.config.redaction_retention_period
+        )
+
+        # We fetch all redactions that point to an event that we have that has
+        # a stream ordering from over a month ago, that we haven't yet censored
+        # in the DB.
+        sql = """
+            SELECT er.event_id, redacts FROM redactions
+            INNER JOIN events AS er USING (event_id)
```

Comment:
Maybe that's just me being picky, but I find that using `er` and `eb` as names (for the same table) here makes it quite hard to understand what the query is supposed to do.